### ISSUE TITLE
incorrect behaviour of FloatRangeSlider for small numbers

### ISSUE THREAD

**Issue by gsteele13** [jupyter-widgets/ipywidgets#3437]
## Description

FloatRangeSlider does not work at all for small numbers (like 1e-8)

## Reproduce

```
import ipywidgets as widgets

w = widgets.FloatRangeSlider()
w.min = -1.63e-08
w.max = 11.5e-08
w.step = 0.1e-8
w.value = (1.65e-08,8.20e-08)
w.readout_format = ".2e"
display(w)

w2 = widgets.FloatRangeSlider()
w2.min = -1.63
w2.max = 11.5
w.step = 0.1
w2.value = (1.65,8.20)
w2.readout_format = ".2e"
display(w2)
```
These sliders should look visually the same (aside from readout scale) but do not:

<img width="660" alt="Screenshot 2022-04-12 at 22 08 54" src="https://user-images.githubusercontent.com/16204006/163045074-c968bc40-1e39-40d7-842c-195542a906f0.png">

In particular, small number one does not function at all. 

## Expected behavior

Should work

## Context

- ipywidgets version 7.6.5
- Operating System and version: Linux 205ab214302e 4.15.0-169-generic #177-Ubuntu SMP Thu Feb 3 10:50:38 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
- Browser and version: Chrome 100.0.4896.75

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.
</pre>
</details>

<details><summary>Command Line Output</summary>

Nothing relevant

<pre>
Paste the output from your command line running `jupyter lab` (or `jupyter notebook` if you use notebook) here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>

Nothing relevant

<pre>
Paste the output from your browser Javascript console here.
</pre>
</details>

### If using JupyterLab
- JupyterLab version:

<details><summary>Installed Labextensions</summary>
<pre>
Paste the output from your command line running `jupyter labextension list`.
</pre>
</details>

**Comment by gsteele13** [comment]
also have same problem for FloatSlider:

```
w = widgets.FloatSlider()
w.min = -1.63e-08
w.max = 11.5e-08
w.step = 0.1e-8
w.value = 1.65e-08
w
```

<img width="387" alt="Screenshot 2022-04-12 at 22 18 26" src="https://user-images.githubusercontent.com/16204006/163046600-eeefa1c3-24c2-4511-bf9f-c4ed117282a6.png">

**Comment by jasongrout** [comment]
This is discussed at https://github.com/jupyter-widgets/ipywidgets/issues/259. The short answer is that it is fixed in ipywidgets 8, which will be released soon. You can install the release candidate 8.0.0rc0 with either pip or conda if you wish.

**Comment by jasongrout** [comment]
This is fixed in the upcoming 8.0.
